### PR TITLE
feat(markdown): markdown codeblock and header

### PR DESCRIPTION
**Pull request by viniciusglinden**
Copied from rmarkdown

### CODE REVIEW

File: snippets/markdown.json [hunk 1]
```diff
@@ -1,4 +1,27 @@
 {
+    "YAML Header": {
+        "prefix": "---",
+        "body": [
+            "---",
+            "title: ${1:title}",
+            "author: ${2: author_name}",
+            "date: ${3:\"`r Sys.Date()`\"}",
+            "output: ${5:pdf_document}",
```
Comment (OkelleyDevelopment) [inline snippets/markdown.json:9]: The 5 should be a 4

File: snippets/markdown.json [hunk 2]
```diff
@@ -1,4 +1,27 @@
...
+            "---"
+        ]
+    },
+    "Insert Inline Chunck": {
+        "prefix": "`",
+        "body": "`${0}`"
+    },
```
Comment (OkelleyDevelopment) [inline snippets/markdown.json:16]: We should remove this snippet as it's going to clash with the one below it. Also, if users have an auto-pairing plugin/configuration then it renders this moot.

File: snippets/markdown.json [hunk 3]
```diff
@@ -1,4 +1,27 @@
...
+    "Insert Code Language": {
+        "prefix": "```",
+        "body": ["```${1:lang}", "${0}", "```"]
+    },
+    "Insert Block Equation": {
+        "prefix": "$$",
+        "body": ["\\$$", "${0}", "\\$$"]
+    },
```
Comment (OkelleyDevelopment) [inline snippets/markdown.json:24]: Could this be omitted in favor of users extending the snippets they load when interacting with markdown files? 

(I'm trying to understand your reasoning for the inclusion of this snippet)

**Review by OkelleyDevelopment — CHANGES_REQUESTED** [review]

**Comment by OkelleyDevelopment** [comment]
Any updates on this @viniciusglinden?

**Comment by OkelleyDevelopment** [comment]
Closing this PR since it's inactivate. Feel free to re-open once the comments above are addressed. If you have concerns, feel free to leave a message as well!

**Comment by viniciusglinden** [comment]
Sorry @OkelleyDevelopment I didnt get any emails from your comments! Trying to correct and reopen...

File: snippets/markdown.json (hunk 3 above)
Comment (viniciusglinden) [reply snippets/markdown.json:24]: This is a copy from `rmarkdown.json`. I would suggest: either we introduce this here, or we delete the one in `rmarkdown.json`...

File: snippets/markdown.json (hunk 1 above)
Comment (viniciusglinden) [reply snippets/markdown.json:9]: Corrected

**Pull request closed without merging**